src/backend/admin_ops/__init__.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2023 SUSE LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Literal
+
+import httpx
+from botocore.auth import HmacV1Auth
+from botocore.awsrequest import AWSRequest
+from botocore.credentials import Credentials
+
+from backend.admin_ops.errors import error_from_response
+
+
+def signed_request(
+    *,
+    access: str,
+    secret: str,
+    method: str,
+    url: str,
+    data: Dict[str, Any] | None = None,
+    params: Dict[str, Any] | None = None,
+    headers: Dict[str, Any] | None = None,
+) -> httpx.Request:
+    """
+    Returns a request signed with AWS HMAC v1 Auth (SHA1), which is what is
+    understood by RGW's admin ops authentication.
+    """
+    creds = Credentials(access, secret)
+    awsreq = AWSRequest(
+        method=method, url=url, data=data, params=params, headers=headers
+    )
+    # NOTE(jecluis): it seems that query parameters are not considered for the
+    # signature, unless they are provided via the URL, and always with a '/'
+    # after the base URL:port address. E.g., http://foo.bar:123/?param1=baz
+    # works, whereas http://foo.bar:123?param1=baz does not.
+    #
+    HmacV1Auth(credentials=creds).add_auth(awsreq)
```

Comment:
It is not clear to me why it is needed here to use a different way to sign the requests instead of using the `Config` class to instantiate a boto3 client.
If the need here is to sign the request using a `v2` signing method, this can be achieved specifying:

`signature_version = 'v2'` 

in the Config object used to instantiate the client.
Unless there are specific reasons to not use the standard boto3 way to authenticate the requests we should prefer the boto3 functionalities provided out of the box. 
Going with a "custom" implementation is not optimal if we have the standard boto3 functionalities to use.

Reply:
See, first of all, it's not a v2 signature - it's a v1, which I thought the comments were particularly clear about -, and I could not find a way to use that other than relying on boto3's internal classes -- note that the `HmacV1Auth` class is not a custom implementation, it is what is offered by boto.

Additionally, yes, this is a custom API, and that's not something boto does.

All we are doing is sending a boto request using a different HHTP library, adjusting our request to the needs of the API and the protocol.

Should you find a better approach though, feel free to let me know and we can adjust it.

Reply:
It seems that with old versions of boto was actually possible to specify custom http calls: https://www.adamsmith.haus/python/docs/boto.connection.AWSAuthConnection

via `build_base_http_request` method.

Unfortunately, this seems not be possible anymore with boto3.

Out of curiosity: why are we using `v1` signature and not `v2`?

Reply:
I could be wrong, but It should work also with v2.